### PR TITLE
Make ironic environment variables configurable

### PR DESCRIPTION
**Pull request by furkatgofurov7**
Fixes [metal3-dev-env/issues/443](https://github.com/metal3-io/metal3-dev-env/issues/443)

### CODE REVIEW

File: tools/run_local_ironic.sh [hunk 1]
```diff
@@ -2,14 +2,34 @@
...
 IRONIC_IMAGE=${IRONIC_IMAGE:-"quay.io/metal3-io/ironic:master"}
 IRONIC_INSPECTOR_IMAGE=${IRONIC_INSPECTOR_IMAGE:-"quay.io/metal3-io/ironic-inspector"}
 IRONIC_KEEPALIVED_IMAGE=${IRONIC_KEEPALIVED_IMAGE:-"quay.io/metal3-io/keepalived"}
 IPA_DOWNLOADER_IMAGE=${IPA_DOWNLOADER_IMAGE:-"quay.io/metal3-io/ironic-ipa-downloader:master"}
 IRONIC_DATA_DIR=${IRONIC_DATA_DIR:-"/opt/metal3-dev-env/ironic"}
 CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
+HTTP_PORT="${HTTP_PORT:-6180}"
+PROVISIONING_INTERFACE="${PROVISIONING_INTERFACE:-provisioning}"
+DHCP_RANGE="${DHCP_RANGE:-172.22.0.10,172.22.0.100}"
+DEPLOY_KERNEL_URL="${DEPLOY_KERNEL_URL:-http://172.22.0.1:6180/images/ironic-python-agent.kernel}"
```
Comment (dhellmann) [inline tools/run_local_ironic.sh:14]: The defaults for these URL variables should probably refer to the HTTP_PORT variable so it's possible to change the port without having to re-define the URLs, too.

File: tools/run_local_ironic.sh [hunk 2]
```diff
@@ -2,14 +2,34 @@
...
+DEPLOY_RAMDISK_URL="${DEPLOY_RAMDISK_URL:-http://172.22.0.1:6180/images/ironic-python-agent.initramfs}"
+IRONIC_ENDPOINT="${IRONIC_ENDPOINT:-http://172.22.0.1:6385/v1/}"
+IRONIC_INSPECTOR_ENDPOINT="${IRONIC_INSPECTOR_ENDPOINT:-http://172.22.0.1:5050/v1/}"
+CACHEURL="${CACHEURL:-http://172.22.0.1/images}"
+IRONIC_FAST_TRACK="${IRONIC_FAST_TRACK:-false}"
+IRONIC_ENV_DATA_DIR=${IRONIC_ENV_DATA_DIR:-"/opt/metal3-dev-env/ironic-vars"}
+
+cat <<EOF > "${IRONIC_ENV_DATA_DIR}"
```
Comment (dhellmann) [inline tools/run_local_ironic.sh:22]: The variable says "DIR" but this creates a file, doesn't it?

File: tools/run_local_ironic.sh [hunk 3]
```diff
@@ -2,14 +2,34 @@
...
+HTTP_PORT="${HTTP_PORT:-6180}"
+PROVISIONING_INTERFACE="${PROVISIONING_INTERFACE:-provisioning}"
+DHCP_RANGE="${DHCP_RANGE:-172.22.0.10,172.22.0.100}"
+DEPLOY_KERNEL_URL="${DEPLOY_KERNEL_URL:-http://172.22.0.1:6180/images/ironic-python-agent.kernel}"
+DEPLOY_RAMDISK_URL="${DEPLOY_RAMDISK_URL:-http://172.22.0.1:6180/images/ironic-python-agent.initramfs}"
+IRONIC_ENDPOINT="${IRONIC_ENDPOINT:-http://172.22.0.1:6385/v1/}"
+IRONIC_INSPECTOR_ENDPOINT="${IRONIC_INSPECTOR_ENDPOINT:-http://172.22.0.1:5050/v1/}"
+CACHEURL="${CACHEURL:-http://172.22.0.1/images}"
+IRONIC_FAST_TRACK="${IRONIC_FAST_TRACK:-false}"
+IRONIC_ENV_DATA_DIR=${IRONIC_ENV_DATA_DIR:-"/opt/metal3-dev-env/ironic-vars"}
```
Comment (fmuyassarov) [inline tools/run_local_ironic.sh:20]: You probably also don't need double quotes around /opt/metal3-dev-env/ironic-vars.

File: tools/run_local_ironic.sh (hunk 1 above)
Comment (furkatgofurov7) [reply tools/run_local_ironic.sh]: Thanks, updated it.

File: tools/run_local_ironic.sh (hunk 2 above)
Comment (furkatgofurov7) [reply tools/run_local_ironic.sh]: Fixed,thanks.

File: tools/run_local_ironic.sh (hunk 3 above)
Comment (furkatgofurov7) [reply tools/run_local_ironic.sh]: Reworked it, thanks.

**Review by maelk — APPROVED** [review]

**Comment by maelk** [comment]
We'll have a bit of a conflict between this PR and https://github.com/metal3-io/baremetal-operator/pull/631 . Hopefully this one will go in first, then I'll adapt the second.

File: tools/run_local_ironic.sh [hunk 4]
```diff
@@ -2,14 +2,37 @@
...
 IRONIC_KEEPALIVED_IMAGE=${IRONIC_KEEPALIVED_IMAGE:-"quay.io/metal3-io/keepalived"}
 IPA_DOWNLOADER_IMAGE=${IPA_DOWNLOADER_IMAGE:-"quay.io/metal3-io/ironic-ipa-downloader:master"}
 IRONIC_DATA_DIR=${IRONIC_DATA_DIR:-"/opt/metal3-dev-env/ironic"}
 CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
+HTTP_PORT="${HTTP_PORT:-6180}"
+PROVISIONING_IP="${PROVISIONING_IP:-172.22.0.2}"
+PROVISIONING_CIDR="${PROVISIONING_CIDR:-24}"
+PROVISIONING_INTERFACE="${PROVISIONING_INTERFACE:-ironicendpoint}"
+DHCP_RANGE="${DHCP_RANGE:-172.22.0.10,172.22.0.100}"
+DEPLOY_KERNEL_URL="${DEPLOY_KERNEL_URL:-http://172.22.0.2:${HTTP_PORT}/images/ironic-python-agent.kernel}"
```
Comment (zaneb) [inline tools/run_local_ironic.sh:16]: Shouldn't the IP address default to `${PROVISIONING_IP}`, not `172.22.0.2`?

In fact, when would you ever need to customise beyond that?

Comment (furkatgofurov7) [reply tools/run_local_ironic.sh]: Hi, thanks for the comment @zaneb ! I think `${PROVISIONING_IP}` is the same as the `.2` IP as of this 
 [env-file](https://github.com/metal3-io/baremetal-operator/blob/178a7754ea52378a4aae0b66c66bff1865d2ecc8/deploy/ironic_ci.env#L2)?

**Comment by furkatgofurov7** [comment]
> We'll have a bit of a conflict between this PR and #631 . Hopefully this one will go in first, then I'll adapt the second.

@maelk this one is ready to go, but there will a bit of conflict with #631 as you said, sorry for that.

/assign @dhellmann 
/cc @zaneb

**Comment by dhellmann** [comment]
/approve

I'm happy with this version. I'll leave it open for @zaneb to look at, since he had comments, too.

**Pull request merged**